src/frontend/src/catalog/source_catalog.rs

```diff
@@ -13,15 +13,26 @@
 // limitations under the License.
 
 use risingwave_pb::catalog::source::Info;
-use risingwave_pb::catalog::Source as ProstSource;
-use risingwave_pb::stream_plan::source_node::SourceType;
+use risingwave_pb::catalog::{Source as ProstSource, StreamSourceInfo, TableSourceInfo};
 
 use super::column_catalog::ColumnCatalog;
 use super::{ColumnId, SourceId};
 use crate::WithOptions;
 
 pub const KAFKA_CONNECTOR: &str = "kafka";
 
+#[derive(Clone, Debug)]
+pub enum SourceCatalogInfo {
+    StreamSource(StreamSourceInfo),
+    TableSource(TableSourceInfo),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum SourceCatalogType {
+    Stream,
+    Table,
+}
```

Comment:
Can we totally remove this `enum`, and add two methods to `SourceCatalog` deciding the source type by `matches!(info, SourceCatalogInfo::Table(..))`?

Reply:
like is_table and is_stream?